### DISCUSSION TITLE
breaking (fracturing) elements or breaking meshes (DEM-FEM)

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#17667] [Q&A Modules: General] [answered]
Hi everyone,

In some solid-mechanics simulations, such as the tensile failure of materials due to fracture creation, breaking the finite-element mesh along element edges, or breaking individual elements into multiple pieces, might be an appropriate modelling strategy.   Has anyone done this using MOOSE?   I'm hoping you'll say "yes - this is fully developed in module XXXX, didn't you know?!".

Not only does the underlying finite-element package (libmesh) have to be flexible enough to break elements, but the problem becomes numerically complicated, as contact-detection between broken surfaces bogs the code down (you don't want those freshly-created surfaces just passing through each other!).

An example of a commercial package that does this: https://www.geomechanica.com/software

a

**Comment by hugary1995** [comment, marked as answer]
yes - this is ~fully~ developed in module Xfem, didn't you know?!

**Reply by WilkAndy** [reply]
OMG, you aren't kidding!   I'm so happy.  Now lots of reading for me.

**Reply by hugary1995** [reply]
FYI, @bwspenc and @jiangwen84 are the main points of contact for the XFEM module.

**Comment by Traiwit** [comment]
Hi @WilkAndy @hugary1995 

Since you guys are on this topic, I want to do a mine simulation that has faults in it.
My supervisor suggested using 'Cohesive Zone'  [https://mooseframework.inl.gov/syntax/Modules/TensorMechanics/CohesiveZoneMaster/index.html](url)

when I see that it does not support the 'Large strain' model, is there any way to work around this problem?

Thank you!
Traiwit

**Reply by jiangwen84** [reply]
@Traiwit here is the PR for large deformation CZM. 

https://github.com/idaholab/moose/pull/17157

**Reply by Traiwit** [reply]
Nice! Thank you @jiangwen84

**Comment by WilkAndy** [comment]
When using XFEM to pull apart a material, everything works fine.  But upon recompaction, the fracture surfaces don't "push back" against each other.  Is this correct, or am i driving XFEM incorrectly?

**Reply by jiangwen84** [reply]
> When using XFEM to pull apart a material, everything works fine. But upon recompaction, the fracture surfaces don't "push back" against each other. Is this correct, or am i driving XFEM incorrectly?

You  can implement a contact constrain on the fracture surfaces.

**Reply by WilkAndy** [reply]
Where is there an example of this?

**Reply by jiangwen84** [reply]
> Where is there an example of this?

We do not have an example in MOOSE. The basic idea is to use `ElemElemConstraint` to enforce contact.  One example to enforce glued contact at fracture surfaces using a penalty method is  `XFEMSingleVariableConstraint`. For more complicated contact constraint, stateful properties at interfaces are usually needed and  https://github.com/idaholab/moose/pull/16891 will address that.

**Reply by WilkAndy** [reply]
I tried `XFEMSingleVariableConstraint` but putting in a few `if (_u[_qp] - _u_neighbor[_qp] > 0)` conditions so the penalty didn't try to glue the surfaces together.  That won't work if the surfaces move too much relative to each other (and setting the `variable` in that object seems only possible in very simple situations).

I looked at the Contact module, but i believe you have to pre-specify the surfaces, which is not what i want.

Any ideas?

**Reply by WilkAndy** [reply]
Oh, sorry, i didn't see your reply @jiangwen84 .  i don't want to glue the faces together.  i want to prevent interpenetration.

**Reply by jiangwen84** [reply]
> Oh, sorry, i didn't see your reply @jiangwen84 . i don't want to glue the faces together. i want to prevent interpenetration.

`XFEM` won't work naturally work with `Contact` module, but we can possibly reuse some existing contact codes. You are correct that we always assume the constraint is applied on a pair of element and large sliding and opening might be a problem.

**Reply by WilkAndy** [reply]
OK, this is good to know.  I think that contact detection throughout the mesh is quite a significant addition to MOOSE, and one that i currently don't have time to contribute to.  It will likely involve fairly cumbersome checks involving looking at what's happening on other processors, in the same way that DEM codes are hard to parallelise using MPI.  So, i think for now the XFEM/Contact stuff won't really work for me, but it's good to know what XFEM is after seeing 1E6 emails about it!
